test_utils/test_utils.go

```diff
@@ -0,0 +1,375 @@
+package test_utils
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/activations"
+	"github.com/onflow/cadence/common"
+	"github.com/onflow/cadence/interpreter"
+	"github.com/onflow/cadence/pretty"
+	"github.com/onflow/cadence/sema"
+	"github.com/onflow/cadence/stdlib"
+	"github.com/onflow/cadence/test_utils/sema_utils"
+
+	"github.com/onflow/cadence/bbq/vm"
+	compilerUtils "github.com/onflow/cadence/bbq/vm/test"
+)
+
+type ParseCheckAndInterpretOptions struct {
+	Config             *interpreter.Config
+	CheckerConfig      *sema.Config
+	HandleCheckerError func(error)
+}
+
+var compile = false
+
+func init() {
+	// TODO: Set the env-variable and re-run the tests.
+	// compile = os.Getenv("COMPILE") == "true"
+}
```

Comment:
This is disabled by default, for now. I'll open a follow-up PR to enable them

Reply:
We can maybe make this a flag, similar to how we have for the smoke tests (see `runSmokeTests`)

Reply:
For the moment, maybe we can use the flag, but in the makefile, we would have to enable each package separately. This might also makes sense, since we don't want to run all the tests with the compiler (e.g: type checker tests, tools-tests, etc.). Will update and see.

Reply:
Added a temporary solution in https://github.com/onflow/cadence/pull/3813/commits/ecdbe8e3a484ed1d2f9f8c5623d2fb17d8f198b5